tests/components/zwave_me/test_config_flow.py

```diff
@@ -0,0 +1,140 @@
+    else:
+        return None
+
+
+@patch("homeassistant.components.zwave_me.get_uuid", mock_uuid)
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    with patch(
+        "homeassistant.components.zwave_me.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["errors"] == {}
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "url": "192.168.1.14",
+                "token": "test-token",
+            },
+        )
+        await hass.async_block_till_done()
+        assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result2["title"] == "ws://192.168.1.14"
+        assert result2["data"] == {
+            "url": "ws://192.168.1.14",
+            "token": "test-token",
+        }
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+@patch("homeassistant.components.zwave_me.get_uuid", mock_uuid)
+async def test_zeroconf(hass: HomeAssistant):
+    """Test starting a flow from zeroconf."""
+    with patch(
+        "homeassistant.components.zwave_me.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result: FlowResult = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_ZEROCONF},
+            data=MOCK_ZEROCONF_DATA,
+        )
+        await hass.async_block_till_done()
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "token": "test-token",
+            },
+        )
+        await hass.async_block_till_done()
+        assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result2["title"] == "ws://192.168.1.14"
+        assert result2["data"] == {
+            "url": "ws://192.168.1.14",
+            "token": "test-token",
+        }
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+@patch("homeassistant.components.zwave_me.get_uuid", mock_uuid)
+async def test_error_handling_zeroconf(hass: HomeAssistant):
+    """Test getting proper errors from no uuid."""
+    data = MOCK_ZEROCONF_DATA
+    data.host = "ws://192.12.12.12"
+    with patch(
+        "homeassistant.components.zwave_me.async_setup_entry",
+        return_value=True,
+    ):
+        result: FlowResult = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_ZEROCONF},
+            data=data,
+        )
+        assert result["type"] == RESULT_TYPE_ABORT
+        assert result["reason"] == "no_valid_uuid_set"
+        await hass.async_block_till_done()
```

Comment:
We only need to block till done if there are scheduled tasks that we want to wait for.

Eg when finishing a successful flow with create entry result we want to block till done to make sure the integration setup task is done within the patch context manager. Otherwise we don't need to block till done.
```suggestion
```